### PR TITLE
Added documentation for plugin manifest and plugin code architecture

### PR DESCRIPTION
**Pull request by krisbitney**

### CODE REVIEW

File: src/docs/quick-start/create-plugin-wrappers/plugin-manifest.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+<TabItem value="example">
+
+```yaml
+format: 0.1.0
+name: MyPlugin
+language: plugin/typescript
+module: ./src/index.ts
+schema: ./src/schema.graphql
+import_redirects:
+  - uri: w3://ens/imported-plugin.eth
```
Comment (cbrzn) [inline src/docs/quick-start/create-plugin-wrappers/plugin-manifest.md:49]:
```suggestion
  - uri: wrap://ens/imported-plugin.eth
```

File: src/docs/quick-start/create-plugin-wrappers/plugin-manifest.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+{label: 'Example', value: 'example'},
+]}>
+<TabItem value="schema">
+
+```yaml
+format: # The manifest format version
+name: # (Optional) Name of plugin
+language: # Plugin programming language
+module: # Entry file 
+schema: # GraphQL schema
```
Comment (cbrzn) [inline src/docs/quick-start/create-plugin-wrappers/plugin-manifest.md:33]: Should we remove the GraphQL wording from our docs? :thinking: Not sure which path we should take here tbh

Comment (krisbitney) [reply src/docs/quick-start/create-plugin-wrappers/plugin-manifest.md:33]: I changed it to "Wrapper schema"

File: src/docs/quick-start/create-plugin-wrappers/plugin-architecture.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+Still, there are some simple requirements and best practices that may guide its code structure.
+
+## Required Exports
+
+A plugin must export a `PluginFactory` to be used by the Polywrap Client.
+A `PluginFactory` is a function that returns a `PluginPackageManifest` and a `factory` method that generates an instance of the plugin.
+
+## TypeScript Support
+
+The Polywrap CLI's [`codegen`](../../reference/cli/commands/codegen) and [`build`](../../reference/cli/commands/build) commands will generate a `wrap` folder containing useful types.
```
[the remark this answers is not in the document]
Comment (krisbitney) [reply src/docs/quick-start/create-plugin-wrappers/plugin-architecture.md:16]: you're 100% correct. I fixed it. Thanks!

**Review by dOrgJelli — APPROVED** [review]

**Pull request merged**